### PR TITLE
Move DAGs table to a reusable DataTable component

### PR DESCRIPTION
**Pull request by DjVinnii**
Move the DAGs Table to a reusable DataTable ui component that can be used for other tables as well.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by DjVinnii** [comment]
By the way, I noticed coverage is not setup yet for the new UI. Any plans on doing this soon?

**Comment by pierrejeambrun** [comment]
I don't think we planned on adding coverage for front-end tests in a near future but that might be a great addition indeed.


There are some static checks failure to fix. Also you can install those checks locally with pre-commit hook to help catch those earlier in the development phase and not have to wait for the CI. You can find more information in https://github.com/apache/airflow/blob/main/contributing-docs/03_contributors_quick_start.rst#configuring-pre-commit

**Comment by DjVinnii** [comment]
Hi @pierrejeambrun! Thanks for the quick comment. The checks should be passing now, lets see if the CI succeeds.


> I don't think we planned on adding coverage for front-end tests in a near future but that might be a great addition indeed!

I'll try to implement this in the coming days, I think it is a good addition to have.

File: airflow/ui/src/components/DataTable.tsx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+"use client";
```
Comment (bbovenzi) [inline airflow/ui/src/components/DataTable.tsx:20]:
```suggestion
```

We don't need this. None of the UI is SSR.

**Review by bbovenzi — APPROVED** [review]
lgtm. just one nit.

**Pull request merged**